### ISSUE TITLE
Podcast - Issue with renaming

### ISSUE THREAD

**Issue by Nicocys** [ampache/ampache#1790]
# Environment

 * Ampache version: 3.8.9
 * Web server + version: 2.4.35
 * Server operating system: ArchLinux
 * Client operating system: ArchLinux

 * Client type:
   - Web

# Settings
Standard settings, nothing fancy. delete_from_disk is set to true, but was set that way after the issue appeared.

# Description

The bug appeared after renaming one podcast: in the Podcast section, clicking on the gear and changing the podcast Title field.
Once this is done, it appears that no folder is moved in the podcast directory set with the catalog, and no new folder is created with the new title as name.
Hence the download of new episodes fails when trying to sync the podcast to get new episodes. Didn't try reading old, previously downloaded ones.

In the logs provided below, "Quick Load" is the new name, old one was much longer. No folder corresponding to it was created by Ampache. The issue was solved by manually renaming the old folder, making sure to keep proper permissions/ownership.

# Logs

2018-10-16 08:48:10 [Nicocys] (podcast_episode) -> Downloading http://feeds.soundcloud.com/stream/450276906-quickload-quickload-17.mp3 to /home/Podcasts/Quick Load/918-450276906-quickload-quickload-17.mp3 ... 
2018-10-16 08:48:11 [ampache] (PHP) -> [Runtime Error] file_put_contents(/home/Podcasts/Quick Load/918-450276906-quickload-quickload-17.mp3): failed to open stream: No such file or directory in file /usr/share/webapps/ampache/lib/class/podcast_episode.class.php(395) 
2018-10-16 08:48:11 [Nicocys] (podcast_episode) -> Error when downloading podcast episode.

**Comment by lachlan-00** [comment]
Closing, I've added a check into $podcast->get_root_path() to create directories if the folder doesn't exist.

This is in my branch and will merge into core soon.
